src/exercise/06.js

```diff
@@ -4,27 +4,35 @@
 import * as React from 'react'
 
 function UsernameForm({onSubmitUsername}) {
-  // 🐨 add a submit event handler here (`handleSubmit`).
-  // 💰 Make sure to accept the `event` as an argument and call
-  // `event.preventDefault()` to prevent the default behavior of form submit
-  // events (which refreshes the page).
-  //
-  // 🐨 get the value from the username input (using whichever method
-  // you prefer from the options mentioned in the instructions)
-  // 💰 For example: event.target.elements[0].value
-  // 🐨 Call `onSubmitUsername` with the value of the input
+  const [username, setUsername] = React.useState('')
+  const [error, setError] = React.useState('')
 
-  // 🐨 add the onSubmit handler to the <form> below
+  function handleSubmit(event){
+    event.preventDefault();
+    onSubmitUsername( username );
+  }
+
+  function handleChange(event){
+    const {value} = event.target
+    const isLowerCase = value === value.toLowerCase()
+    if ( ! isLowerCase ) {
+      setError('Username must be lower case')
+    } else { 
+      setError(null)
+    }
+    setUsername(value)
+  }
 
-  // 🐨 make sure to associate the label to the input.
-  // to do so, set the value of 'htmlFor' prop of the label to the id of input
   return (
-    <form>
+    <form onSubmit={handleSubmit}>
       <div>
-        <label>Username:</label>
-        <input type="text" />
+        <label htmlFor="username">Username:</label>
+        <input type="text" id="username" value={username} onChange={handleChange} />
+        <div role="alert" style={{color: 'red'}}>
+          {error}
+        </div>
       </div>
-      <button type="submit">Submit</button>
+      <button disabled={Boolean(error)} type="submit">Submit</button>
```

Comment:
is this with the extra cred? iirc we removed the error handling eventually

Reply:
Yeah I combined some things to make it more fun, so you can't submit if there is a uppercase letter but it won't remove the text either so you can correct it.